include/seastar/parquet/file_reader.hh

```diff
@@ -0,0 +1,475 @@
+#pragma once
+
+#include <seastar/parquet/exception.hh>
+#include <seastar/parquet/schema.hh>
+#include <seastar/parquet/parquet_types.h>
+#include <seastar/parquet/rle_encoding.hh>
+#include <seastar/parquet/overloaded.hh>
+
+#include <seastar/core/file.hh>
+#include <seastar/core/fstream.hh>
+#include <seastar/core/seastar.hh>
+
+#include <optional>
+#include <variant>
+
+namespace parquet {
+
+////////////////////////////////////////////////////////////////////////////////
+
+/* A dynamically sized buffer. Rounds up the size given in constructor to a power of 2.
+ */
+class buffer {
+    size_t _size;
+    std::unique_ptr<uint8_t[]> _data;
+    static constexpr inline uint64_t next_power_of_2(uint64_t n) {
+        if (n < 2) return n;
+        return 1ull << (64 - __builtin_clzll(n - 1));
```

Comment:
Take a look at `include/seastar/core/bitops.hh` - we have wrappers for clz (`count_leading_zeros` and also `log2ceil` and `log2floor`, which is precisely what you can apply here), and it's preferable to use them, in case we want to support different architectures in the future.

Reply:
Fixed in f38fe11.
> it's preferable to use them, in case we want to support different architectures in the future

Yeah, sure...